rebasehelper/build_log_hooks/files_build_log_hook.py

```diff
@@ -0,0 +1,241 @@
+                    error_type = 'missing'
+                elif error_type == 'missing' and match.group('path'):
+                    files.add(match.group('path'))
+                elif error_type and not match.group('path'):
+                    break
+
+        return error_type, list(files)
+
+    @classmethod
+    def _get_best_matching_files_section(cls, rebase_spec_file, file):
+        """Finds a %files section with a file that has the closest match with
+        the specified file.
+
+        Args:
+            rebase_spec_file (specfile.SpecFile): Rebased SpecFile object.
+            file (str): Path to the file to be classified.
+
+        Returns:
+            str: Name of the section containing the closest matching file.
+
+        """
+        best_match = ''
+        best_match_section = ''
+        for sec_name, sec_content in six.iteritems(rebase_spec_file.spec_content.sections):
+            if sec_name.startswith('%files'):
+                for line in sec_content:
+                    new_best_match = difflib.get_close_matches(file, [best_match, MacroHelper.expand(line)])
+                    if new_best_match:
+                        # the new match is a closer match
+                        if new_best_match[0] != best_match:
+                            best_match = new_best_match[0]
+                            best_match_section = sec_name
+
+        return best_match_section or rebase_spec_file.get_main_files_section()
+
+    @classmethod
+    def _correct_missing_files(cls, rebase_spec_file, files):
+        """Adds files found in sources to the SPEC file."""
+        macros = [m for m in rebase_spec_file.macros if m['name'] in MacroHelper.MACROS_WHITELIST]
+        macros = MacroHelper.expand_macros(macros)
+        # ensure maximal greediness
+        macros.sort(key=lambda k: len(k['value']), reverse=True)
+
+        for file in files:
+            section = cls._get_best_matching_files_section(rebase_spec_file, file)
+            substituted_path = MacroHelper.substitute_path_with_macros(file, macros)
+            rebase_spec_file.spec_content.sections[section].insert(0, substituted_path)
+            logger.info('Added %s to %s section', substituted_path, section)
+
+    @classmethod
+    def _correct_deleted_files(cls, rebase_spec_file, files):
+        """Removes files missing in sources from the SPEC file."""
+        for sec_name, sec_content in six.iteritems(rebase_spec_file.spec_content.sections):
+            if sec_name.startswith('%files'):
+                i = 0
+                while i < len(sec_content):
+                    line = sec_content[i]
+                    if line != '':
+                        split_line = line.split(' ')
+                        directives = []
+                        for element in reversed(split_line):
+                            if element in cls.FILES_DIRECTIVES:
+                                if element == '%doc' or element == '%license':
```

Comment:
Since the directives are defined as constants, it doesn't make sense to use some of them directly here. How about adding some property to the constants, specifying their default value?